report/report.go

```diff
@@ -43,6 +43,10 @@ func GenerateReport(r *helpers.DetectionResults, directory string) (path string,
 		jsonFilePath = filepath.Join(path, jsonFileName)
 	}
 
+	if directory == "" {
```

Comment:
Is the idea that talisman reporting will always delete and re-create a folder called `talisman_reports` ?

Reply:
I ask because
```
-r, --reportdirectory string   directory where the scan reports will be stored
-s, --scan                               scanner scans the git commit history for potential secrets
```

Reply:
> 
> 
> Is the idea that talisman reporting will always delete and re-create a folder called `talisman_reports` ?

Yes... if talisman_reports is to be created inside repo directory(UNTRUSTED) then Possibilities like
1. talisman_reports may be a symlink
2. talisman_reports/data may be a symlink
3. talisman_reports/data/report.json may be a symlink
4. There is a **file named talisman_reports** in the repo
5. There is **file** named **data** inside **talisman_reports** directory
6. There is **folder** named **data.json** inside **talisman_reports/data**

These all cases will need to unintentional behavior. Completely Removing **talisman_reports** looked like simplest solution :smile:

Reply:
> 
> 
> I ask because
> 
> ```
> -r, --reportdirectory string   directory where the scan reports will be stored
> -s, --scan                               scanner scans the git commit history for potential secrets
> ```

Completely forgot possibility where --report-directory is relative path inside repo folder itself :sweat_smile: